ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/modalActivityBlock/modalActivityBlock.component.ts

```diff
@@ -0,0 +1,165 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    Output,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { DialogPosition, MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { NoopScrollStrategy } from '@angular/cdk/overlay';
+
+import { of, Observable } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+import { ActivityInstance } from '../../../../models/activityInstance';
+import { ActivityForm } from '../../../../models/activity/activityForm';
+import { ActivityServiceAgent } from '../../../../services/serviceAgents/activityServiceAgent.service';
+
+const DEFAULT_DIALOG_SETTINGS = {
+    hasBackdrop: true,
+    autoFocus: false,
+    scrollStrategy: new NoopScrollStrategy(),
+};
+
+const EDIT_DIALOG_CONFIG: MatDialogConfig = {
+    ...DEFAULT_DIALOG_SETTINGS,
+    width: `862px`,
+    position: {top: '65px'},
+    panelClass: 'modal-activity-block__edit-dialog',
+};
+
+@Component({
+    selector: 'ddp-modal-activity-block',
+    styleUrls: ['modalActivityBlock.component.scss'],
+    templateUrl: 'modalActivityBlock.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ModalActivityBlockComponent {
+    @Input() studyGuid: string;
+    @Input() activityGuid: string;
+    @Input() instance: ActivityInstance;
+    @Input() validationRequested: boolean;
+    @Input() readonly: boolean;
+    @Output() deleteActivity = new EventEmitter<string>();
+
+    @ViewChild('edit_dialog') private editModalRef: TemplateRef<any>;
+    @ViewChild('delete_dialog') private deleteModalRef: TemplateRef<any>;
+    @ViewChild('delete_button', {read: ElementRef}) private deleteButtonRef: ElementRef;
+
+    public activityForm: ActivityForm;
+
+    constructor(private readonly activityServiceAgent: ActivityServiceAgent,
+                private dialog: MatDialog,
+                private cdr: ChangeDetectorRef) {
+    }
+
+    public deleteActivityInstance(): void {
+        this.activityServiceAgent.deleteActivityInstance(
+            this.studyGuid,
+            this.instance.instanceGuid
+            )
+            .pipe(take(1))
+            .subscribe(() => {
+                this.deleteActivity.emit(this.instance.instanceGuid);
+            });
+    }
+
+    public openEditDialog(): void {
+        this.getFullActivity()
+            .then((activity: ActivityForm) => {
+                this.activityForm = activity;
+                this.openDialog(this.editModalRef, EDIT_DIALOG_CONFIG, this.closeEditDialog.bind(this));
+            })
+            .catch((err) => {
+                console.error('An error during getting a full activity', err);
```

Comment:
Please use LoggingService . There are examples of usage in code.

Reply:
Let's avoid using promises! And if we encounter them in some library, let's wrap them with observables!